pkg/ccl/changefeedccl/changefeed_stmt.go

```diff
@@ -348,6 +348,26 @@ func createChangefeedJobRecord(
 		statementTime = initialHighWater
 	}
 
+	// Change the statement time to highwater time if available to make sure
+	// that we do not get an error when getting table descriptors (might happen
+	// in case where cursor time is far behind in the past)
+	var progressHighWater *hlc.Timestamp
+	job, err := p.ExecCfg().JobRegistry.LoadJobWithTxn(ctx, jobID, p.Txn())
```

Comment:
I don't think you need to load the job at all;
Alter changefeed passes an annotatedChangefeedStatemetn containing changefeed statement, along with things that are necessary to make "alter" work.  So, you can include whatever extra information, such as high watermark there.